src/components/map/DynamicMap.tsx

```diff
@@ -0,0 +1,38 @@
+import Leaflet from "leaflet";
+import "leaflet/dist/leaflet.css";
+import { memo, ReactNode, useEffect } from "react";
+import * as ReactLeaflet from "react-leaflet";
+
+interface MapProps {
+  className?: string;
+  children: ReactNode;
+}
+
+const Map = ({ children, className, ...rest }: MapProps) => {
+  useEffect(() => {
+    (async function init() {
+      // @ts-ignore
+      delete Leaflet.Icon.Default.prototype._getIconUrl;
+      Leaflet.Icon.Default.mergeOptions({
+        iconRetinaUrl: "leaflet/images/marker-icon-2x.webp",
+        iconUrl: "leaflet/images/marker-icon.webp",
+        shadowUrl: "leaflet/images/marker-shadow.webp",
+      });
```

Comment:
even if it's IIFE. 
i think it is better wrapping this function in a callback and invoke it in with a better decleration 

(wellNamedFunction)()

Reply:
This file is a duplicate of the `DynamicMap.tsx` file. Let me check the file and change it based on its contents.

Reply:
fixed